### PR TITLE
[script.module.dropbox] 8.4.2

### PR DESCRIPTION
**Pull request by micahg**
### Description
* Fix Windows import errors (@robweber)
* Make qrcode module an optional dependency (@robweber)
### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [script.foo.bar] 1.0.0

### CODE REVIEW

**Comment by micahg** [comment]
Hi, I _think_ the errors picked out by @TravisBuddy are all in the dropbox source (eg: not my code). Are there any issues I _must_ fix to get this merged?

Thanks,

**Comment by Rechi** [comment]
The error is reported for a non upstream file. Warnings and infos are just warnings and infos and do not fail the build.

**Comment by micahg** [comment]
yeah sorry -- i was misreading @TravisBuddy

File: script.module.dropbox/addon.xml [hunk 1]
```diff
@@ -1,33 +1,36 @@
 <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
 <addon id="script.module.dropbox"
        name="Dropbox API"
-       version="8.4.1"
+       version="8.4.2"
        provider-name="micah">
```
Comment (Rechi) [inline script.module.dropbox/addon.xml]: Official author of the Dropbox module is `Dropbox`. https://github.com/dropbox/dropbox-sdk-python/blob/v8.4.1/setup.py#L59.

File: script.module.dropbox/addon.xml [hunk 2]
```diff
@@ -1,33 +1,36 @@
...
   <requires>
     <import addon="xbmc.python" version="2.25.0"/>
     <import addon="script.module.requests" version="2.18.4"/>
```
Comment (Rechi) [inline script.module.dropbox/addon.xml]: upstream requires `requests >= 2.16.2`

File: script.module.dropbox/addon.xml [hunk 3]
```diff
@@ -1,33 +1,36 @@
...
-    <import addon="script.module.qrcode" version="5.2"/> <!-- For QR Code display for Dropbox API authentication -->    
+    <import addon="script.module.six" version="1.11.0" />
```
Comment (Rechi) [inline script.module.dropbox/addon.xml]: upstream requires `six >= 1.3.0`

File: script.module.dropbox/addon.xml [hunk 4]
```diff
@@ -1,33 +1,36 @@
...
+    <import addon="script.module.qrcode" version="5.2" optional="true" /> <!-- For QR Code display for Dropbox API authentication -->
   </requires>
-  <extension point="xbmc.python.library" library="authenticate.py"> <!-- entry to start the authentification script from the settings dialog of other addons -->
-    <provides> </provides> <!-- Addon should not be shown anywhere -->
-  </extension>
+  <extension point="xbmc.python.library" library="authenticate.py" /> <!-- entry to start the authentification script from the settings dialog of other addons -->
   <extension point="xbmc.python.module" library="lib" /> <!-- entry to import this in other addons -->
   <extension point="xbmc.addon.metadata">
     <summary lang="en">Dropbox API</summary>
```
Comment (Rechi) [inline script.module.dropbox/addon.xml]: `lang="en_GB"`

File: script.module.dropbox/addon.xml [hunk 5]
```diff
@@ -1,33 +1,36 @@
...
+    <import addon="script.module.qrcode" version="5.2" optional="true" /> <!-- For QR Code display for Dropbox API authentication -->
   </requires>
-  <extension point="xbmc.python.library" library="authenticate.py"> <!-- entry to start the authentification script from the settings dialog of other addons -->
-    <provides> </provides> <!-- Addon should not be shown anywhere -->
-  </extension>
+  <extension point="xbmc.python.library" library="authenticate.py" /> <!-- entry to start the authentification script from the settings dialog of other addons -->
   <extension point="xbmc.python.module" library="lib" /> <!-- entry to import this in other addons -->
   <extension point="xbmc.addon.metadata">
     <summary lang="en">Dropbox API</summary>        
     <description lang="en">Provide Dropbox Python API support</description>
```
Comment (Rechi) [inline script.module.dropbox/addon.xml]: `lang="en_GB"`

File: script.module.dropbox/addon.xml [hunk 6]
```diff
@@ -1,33 +1,36 @@
...
     <description lang="he">התוסף מספק תמיכה ב-API Python Dropbox</description>
```
Comment (Rechi) [inline script.module.dropbox/addon.xml]: `lang="en_IL"`

File: script.module.dropbox/addon.xml [hunk 7]
```diff
@@ -1,33 +1,36 @@
...
-    <disclaimer lang="en"></disclaimer>     
     <platform>all</platform>
     <language/>
     <license>Apache2</license>
-    <forum></forum>
-    <website></website>
-    <source>http://python-requests.org</source>
+    <website>https://www.dropbox.com</website>
```
Comment (Rechi) [inline script.module.dropbox/addon.xml]:
```suggestion
    <website>https://www.dropbox.com/developers/documentation/python</website>
```

File: script.module.dropbox/addon.xml [hunk 8]
```diff
@@ -1,33 +1,36 @@
...
+    <source>https://www.dropbox.com/developers/documentation/python</source>
```
Comment (Rechi) [inline script.module.dropbox/addon.xml]:
```suggestion
    <source>https://github.com/dropbox/dropbox-sdk-python</source>
```

File: script.module.dropbox/addon.xml [hunk 9]
```diff
@@ -1,33 +1,36 @@
...
     <email>micahgalizia AT gmail.com</email>
```
Comment (Rechi) [inline script.module.dropbox/addon.xml]: upstream email is `dev-platform@dropbox.com`

File: script.module.dropbox/lib/dropbox/pkg_resources.py [hunk 10]
```diff
@@ -0,0 +1,3 @@
+
+def resource_filename(*args):
```
Comment (Rechi) [inline script.module.dropbox/lib/dropbox/pkg_resources.py]: If you want to provide `pkg_resources.resource_filename(...)`, this one has wrong arguments.
Correct is `def resource_filename(package_or_requirement, resource_name):`

File: script.module.dropbox/lib/dropbox/pkg_resources.py [hunk 11]
```diff
@@ -0,0 +1,3 @@
...
+    return "trusted-certs.crt"
```
Comment (Rechi) [inline script.module.dropbox/lib/dropbox/pkg_resources.py]: Why hardcode the return value, instead of implementing it the correct way?

Comment (robweber) [reply script.module.dropbox/lib/dropbox/pkg_resources.py]: These changes came from me. These additions came from [some advice given in the forum](https://forum.kodi.tv/showthread.php?tid=143406&pid=1221401#pid1221401). The dropbox library throws an error when pkg_resources is not present and this was the suggested work around to giving the library what it wanted without a lot of extra work. This function, returning the cert location, is all it cares about and gets the library to load and function. I've been using this workaround as part of the Backup addon for years.

Comment (Rechi) [reply script.module.dropbox/lib/dropbox/pkg_resources.py]: I had the following in mind, which isn't a lot of extra code, but will also work if additional `pkg_resources.resource_filename` calls are added later.
```
def resource_filename(package_or_requirement, resource_name):
    return resource_name
```

Comment (robweber) [reply script.module.dropbox/lib/dropbox/pkg_resources.py]: Makes sense to me. @micahg - do you want me to make this change and do another PR on your repo so you can merge this in? I see there are other changes to consider from above as well. Just let me know if you need more from me.

**Comment by micahg** [comment]
Hey,  I'm good to push up the requested changes -- just want some clarification on the versions above.

Thanks,

File: script.module.dropbox/addon.xml (hunk 2 above)
Comment (Rechi) [reply script.module.dropbox/addon.xml]: It's not a blocker, but I see no reason to not use the upstream requirements.

File: script.module.dropbox/addon.xml (hunk 9 above)
Comment (Rechi) [reply script.module.dropbox/addon.xml]: Usually I would set this to the upstream email, or do not provide the email tag. If you want to keep your own email here, it's fine for me.

**Comment by micahg** [comment]
Hi @Rechi, if its ok with you, I'll leave the versions where they were. I think I've addressed everything else.

Thanks,

File: script.module.dropbox/addon.xml (hunk 6 above)
Comment (Rechi) [reply script.module.dropbox/addon.xml]: still needs to be changed

**Comment by Rechi** [comment]
> Version 8.4.2 should contain upstream code of version 8.4.1 (https://github.com/dropbox/dropbox-sdk-python/releases/tag/v8.4.1). (There is no upstream 8.4.2 version).

still missing

> Each add-on submission should be a single commit with using the following style: [script.foo.bar] 1.0.0

please squash the commits

**Comment by micahg** [comment]
Ah, right 8.4.2 -- how would you prefer me to version it? I can't rerelease 8.4.1 and I don't plan on moving to version 9.

Will squash once the version is sorted.

**Comment by Rechi** [comment]
You can name it 8.4.2, but it has to contain upstream code 8.4.1. script.module.dropbox 8.4.1 also didn't contain 8.4.1 upstream code.

**Comment by micahg** [comment]
Hello @Rechi -- apologies for taking so long to get back to this.

There were actually just a few minor changes between 8.4.0 and 8.4.1. They're merged in here now.

File: script.module.dropbox/addon.xml (hunk 6 above)
Comment (Rechi) [reply script.module.dropbox/addon.xml]: change is still missing

**Comment by micahg** [comment]
OK, sorry I missed that he is now he_IL. Hopefully that's everything now.

**Comment by micahg** [comment]
Thanks for sticking it out @Rechi. Also, thanks for the patches @robweber

**Pull request merged**